proving/proving.go

```diff
@@ -51,29 +32,22 @@ func Generate(ctx context.Context, ch shared.Challenge, cfg config.Config, logge
 		return nil, nil, err
 	}
 
-	result, err := postrs.GenerateProof(options.datadir, ch, cfg)
+	result, err := postrs.GenerateProof(options.datadir, ch, cfg, 20, options.powScrypt)
 	if err != nil {
 		return nil, nil, fmt.Errorf("generating proof: %w", err)
 	}
 	logger.Info("proving: generated proof")
-	logger.Debug("Nonce: %v, Indicies: %v, K2PoW: %d, K3PoW: %d", result.Nonce, result.Indicies, result.K2Pow, result.K3Pow)
-	indicies, err := compressIndicies(result.Indicies, options.numUnits, cfg)
-	if err != nil {
-		return nil, nil, fmt.Errorf("compressing proof indicies: %w", err)
-	}
+	logger.Debug("Nonce: %v, Indices: %v, K2PoW: %d, K3PoW: %d", result.Nonce, result.Indices, result.K2Pow, result.K3Pow)
 
-	proof := &shared.Proof{Nonce: result.Nonce, Indices: indicies, K2Pow: result.K2Pow, K3Pow: result.K3Pow}
+	proof := &shared.Proof{Nonce: result.Nonce, Indices: result.Indices, K2Pow: result.K2Pow, K3Pow: result.K3Pow}
```

Comment:
Indices are not compressed any more? That should make ATXs a lot bigger.

If a node has 1 TB of PoST data, only 40 bits are needed to encode label indices, but the indices are uint64s. So the 24 MSB are always 0. Even 1 PB has 14 bits that are always 0.

ATXs are part of the mesh, so if we can save space there we should imo. We can still do the compression and decompression in go, since it should be much faster than the actual proof generation & verification.

Reply:
Indices are compressed in post-rs already.